taichi/backends/vulkan/aot_module_builder_impl.cpp

```diff
@@ -147,6 +149,10 @@ void AotModuleBuilderImpl::add_per_backend(const std::string &identifier,
   ti_aot_data_.spirv_codes.push_back(compiled.task_spirv_source_codes);
 }
 
+void AotModuleBuilderImpl::add_compiled_kernel(aot::Kernel *kernel) {
+  kernel->save_to_module(this);
```

Comment:
I'm just thinking if we can put `save_to_module`'s impl within each module builder impl. That way `aot::Kernel` doesn't need to know anything about the module.

Furthermore, `aot::Kernel` has a misleading namespace now. It's probably more suitable to call this `CompiledKernel` or something...

Reply:
+1 on the misleading namespace, I'll leave that to a separate search and replace PR!